app/geocode/DynamicMarker.tsx

```diff
@@ -32,35 +52,37 @@ function DynamicMarker({
         ? value
         : Number(value.toString().replace(",", "."));
     },
-    [record, latColumnName, longColumnName],
+    [record],
+  );
+
+  // Calculer lat/long directement depuis les noms de colonnes
+  const lat = useMemo(
+    () => parseGeoCoordinate(latColumnName),
+    [parseGeoCoordinate, latColumnName],
   );
 
-  const [lat, setLat] = useState<number | null>(getGeoAsNumber(true, false));
-  const [long, setLong] = useState<number | null>(getGeoAsNumber(false, true));
+  const long = useMemo(
+    () => parseGeoCoordinate(longColumnName),
+    [parseGeoCoordinate, longColumnName],
+  );
+
+  // Utiliser un ref pour éviter les vols répétés vers les mêmes coordonnées
+  const lastFlyToCoords = useRef<{ lat: number; long: number } | null>(null);
 
+  // useEffect uniquement pour synchroniser avec Leaflet (système externe)
   useEffect(() => {
-    if (record && mappings) {
-      setLatColumnName(String(mappings[COLUMN_MAPPING_NAMES.LATITUDE.name]));
-      setLongColumnName(String(mappings[COLUMN_MAPPING_NAMES.LONGITUDE.name]));
-      setNormAddressColumnName(
-        String(mappings[COLUMN_MAPPING_NAMES.NORMALIZED_ADDRESS.name]),
-      );
-      setLat(getGeoAsNumber(true, false));
-      setLong(getGeoAsNumber(false, true));
-      if (lat && long && lat !== 0 && long !== 0) {
-        map.flyTo([lat, long]);
-      }
+    if (
+      lat &&
+      long &&
+      lat !== 0 &&
+      long !== 0 &&
```

Comment:
if `lat` and `long` equal to `0`, they should be falsy. So I think these line could be removed.

```suggestion
```